client/dashboard/sites/hosting-feature-activation-modal/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { siteAutomatedTransfersEligibilityQuery } from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import { __experimentalVStack as VStack } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '../../components/notice';
+import { DataCenterForm } from './data-center-form';
+import { hasBlockingError, ErrorContentInfo } from './error-content-info';
+import { WarningContentInfo } from './warning-content-info';
+
+export default function HostingFeatureActivationModal( { siteId }: { siteId: number } ) {
+	const { data } = useQuery( siteAutomatedTransfersEligibilityQuery( siteId ) );
+	if ( ! data ) {
+		return null;
```

Comment:
If you're going to render nothing during loading, then I think a better strategy is to load the AT eligibility query using a `useSuspenseQuery` and render the modal like

```ts
<Suspense fallback={ null }>
	<Modal
		title={ __( 'Before you continue' ) }
		onRequestClose={ () => setIsModalOpen( false ) }
		size="medium"
	>
		<HostingFeatureActivationModal siteId={ site.ID } />
	</Modal>
</Suspense>
```

That way it doesn't flash up a small white modal frame, only for it then to grow into a large which modal frame a second later.

Reply:
Good point, updated in e6bab9036394006499ce215127d77f7b10a84e9e.